src/libfetchers/git-utils.cc

```diff
@@ -226,13 +244,10 @@ struct GitRepoImpl : GitRepo, std::enable_shared_from_this<GitRepoImpl>
     {
         initLibGit2();
 
-        if (pathExists(path.string())) {
-            if (git_repository_open(Setter(repo), path.string().c_str()))
-                throw Error("opening Git repository '%s': %s", path, git_error_last()->message);
-        } else {
-            if (git_repository_init(Setter(repo), path.string().c_str(), bare))
-                throw Error("creating Git repository '%s': %s", path, git_error_last()->message);
-        }
+        if (!pathExists(path.string()))
+            initRepoAtomically(path, bare);
+        if (git_repository_open(Setter(repo), path.string().c_str()))
+            throw Error("opening Git repository %s: %s", path, git_error_last()->message);
```

Comment:
```suggestion
            throw Error("opening Git repository '%s': %s", path, git_error_last()->message);
```

Reply:
Check the error message above. This gets quoted twice. Is this the case for all nix::Path the case?

Reply:
Oh you're right, `std::filesystem::path` behaves differently than `Path` (i.e. `std::string`).

Reply:
So I believe we are good?